tests/core/test_solve.py

```diff
@@ -1252,18 +1249,12 @@ def test_fast_update_with_update_modifier_not_set():
         pprint(convert_to_dist_str(unlink_precs))
         pprint(convert_to_dist_str(link_precs))
         unlink_order = (
-            'channel-4::python-2.7.14-h89e7a4a_22',
             'channel-4::sqlite-3.21.0-h1bed415_2',
-            'channel-4::libedit-3.1-heed3624_0',
             'channel-4::openssl-1.0.2l-h077ae2c_5',
-            'channel-4::ncurses-6.0-h9df7e31_2',
         )
         link_order = (
-            'channel-4::ncurses-6.1-hf484d3e_0',
             'channel-4::openssl-1.0.2p-h14c3975_0',
-            'channel-4::libedit-3.1.20170329-h6b74fdf_2',
```

Comment:
This mess is stuff I messed up in earlier 4.7.  This PR restores some sanity here by doing a better job with knowing when to freeze specs.